src/textual/_node_list.py

```diff
@@ -63,7 +63,7 @@ def index(self, widget: Widget) -> int:
         """
         return self._nodes.index(widget)
 
-    def _get_by_id(self, widget_id: str) -> Widget | None:
```

Comment:
What was PyLance complaining about?

Reply:
When used in `Widget.get_child_by_id`, it shows:
> "_get_by_id" is protected and used outside of the class in which it is declared

Reply:
Ah.

PyLance is using an overly strict interpretation of leading underscores there. Which doesn’t follow from pep8. Actually are you sure that’s PyLance and not a linter? I didn’t recall seeing any red underlines in vscode.

We usd leading underscores to mean “not part of the public API”. I don’t think it makes sense for this method to be public.

Reply:
Yeah it gives me this link to `pyright`: https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportPrivateUsage

Understood on the different interpretations, I'll change it back!

Reply:
I do have __Type Checking Mode__ set to `strict`, maybe that's what did it.